### PR TITLE
Sign windows executable

### LINKED ISSUE #12: Sign Windows CLI executable
https://learn.microsoft.com/en-us/windows/win32/seccrypto/signtool
https://www.microsoft.com/en-us/wdsi/filesubmission
https://github.com/azure/trusted-signing-action

- [x] create Trusted Signer in Azure portal
- [x] create a Verified Identity in Azure portal
- [x] create a Certificate Policy in Azure portal

The tricky part is that MacOS notarization requires a MacOS runner and Windows signing requires a Windows runner, but both are currently done by GoReleaser, which runs in a single step (on a single runner.) We'll have to split this existing GoReleaser manifest in 2 (or 3) parts so we can have one part done on a MacOS runner and one part done on a Windows runner. See https://github.com/goreleaser/goreleaser-pro-prebuilt-example?tab=readme-ov-file

### PR DESCRIPTION
**Pull request by lionello**
Fixes #12 

Reqs:
- MacOS: codesign an notarytool
- Windows: ~~signtool~~ using action
- Nix: nix-prefetch-url (required during the release step)

### CODE REVIEW

**Comment by lionello** [comment]
FYI, GoReleaser now supports MacOS signing and notarization (which would replace the shell scripts we currently use): https://goreleaser.com/customization/notarize/

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -135,36 +137,148 @@ jobs:
...
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Trusted Signing
+        uses: Azure/trusted-signing-action@v0.3.20
+        if: startsWith(github.ref, 'refs/tags/v') # only run this step on tagged commits
+        with:
+          endpoint: https://wus2.codesigning.azure.net/ # from Azure portal
+          trusted-signing-account-name: DefangLabs # from Azure portal
+          certificate-profile-name: signed-binary-test # from Azure portal
```
Comment (edwardrf) [inline .github/workflows/go.yml:163]: I presume this name is not publicly viewable or be any part of the cert, or the cert is not of any testing nature.

**Review by edwardrf — APPROVED** [review]
LGTM

File: .github/workflows/go.yml (hunk 1 above)
Comment (lionello) [reply .github/workflows/go.yml:163]: The cert is managed in Azure and will never be exposed. All signing happens in Azure.

**Pull request merged**